lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,690 @@
+application. We write our tests with relative URLs because L<Test::Mojo> takes
+care of getting the request to the running test application (since its port
+may change between runs).
+
+=head3 Testing with configuration data
+
+We can alter the behavior of our application using environment variables (such
+as C<MOJO_MODE>) and through configuration values. One nice feature of
+L<Test::Mojo> is its ability to pass configuration values directly from its
+constructor.
+
+Let's modify our application and add a "feature flag" to enable a new feature
+when the C<enable_weather> configuration value is set:
+
+  # Load configuration from hash returned by "my_app.conf"
+  my $config = $self->plugin('Config');
+
+  # Normal route to controller
+  $r->get('/')->to('example#welcome');
+
+  # NEW: this route only exists when 'enable_weather' is set in the configuration
+  if ($config->{enable_weather}) {
+      $r->get('/weather')->to(cb => sub { shift->render(text => "It's hot! 🔥") }
+  }
+
+To test this new feature, we don't even need to create a configuration file—we
+can simply pass the configuration data to the application directly via
+L<Test::Mojo>'s constructor:
+
+  my $t = Test::Mojo->new('MyApp', {enable_weather => 1});
+  $t->get_ok('/')->status_is(200)->content_like(qr/Mojolicious/i);
+  $t->get_ok('/weather')->status_is(200)->content_like(qr/🔥/);
+
+When we run these tests, L<Test::Mojo> will pass this configuration data to
+the application, which will cause it to create a special C</weather> route
+that we can access in our tests. Unless C<enable_weather> is set in a
+configuration file, this route will not exist when the application
+runs. Feature flags like this allow us to do soft rollouts of features,
+targeting a small audience for a period of time. Once the feature has been
+proven, we can refactor the conditional and make it a full release.
+
+This example shows how easy it is to start testing a Mojolicious application
+and how to set specific application configuration directives from a test file.
+
+=head3 Testing application helpers
+
+Let's say we register a helper in our application to generate an HTTP Basic
+Auth header value:
+
+  use Mojo::Util 'b64_encode';
+
+  app->helper(basic_auth => sub {
+    my $c = shift;
+    return {Authorization => 'Basic ' . b64_encode join(':' => @_), ''};
+  });
+
+How do we test application helpers like this? L<Test::Mojo> has access to the
+application object, which allows us to invoke helpers from our test file:
+
+  my $t = Test::Mojo->new('MyApp');
+
+  is_deeply $t->app->basic_auth(bif => "Bif's Passwerdd"),
+    {Authorization => 'Basic YmlmOkJpZidzIFBhc3N3ZXJkZA=='},
+    'correct header value';
+
+Any aspect of the application (helpers, plugins, routes, etc.) can be
+introspected from L<Test::Mojo> through the application object. This enables
+us to get deep test coverage of L<Mojolicious>-based applications.
+
+=head1 ASSERTIONS
```

Comment:
You're doing great, i'm having a hard time finding new things to nitpick about. 😉